unumpy/_multimethods.py

```diff
@@ -973,20 +969,22 @@ def average(a, axis=None, weights=None, returned=False):
 
 
 def _mean_default(a, axis=None, dtype=None, out=None, keepdims=False):
-    a, dims = _ureduce(a, axis)
-
     if dtype is None:
         if a.dtype.type == "i":
             dtype = float
         else:
             dtype = a.dtype
 
-    N = a.shape[-1]
+    if axis is None:
+        axis = tuple(range(len(a.shape)))
+    if not isinstance(axis, collections.abc.Sequence):
+        axis = (axis,)
 
-    a = sum(a, axis=-1, dtype=dtype) / N
+    N = 0
+    for ax in axis:
+        N += a.shape[ax]
```

Comment:
```suggestion
    N = 1
    for ax in axis:
        N *= a.shape[ax]
```

Reply:
Also, you might want a `_prod` helper so you can do:
```
N = _prod(a.shape[ax] for ax in axis)
```
Since this comes up in a few different places.

Reply:
This is fixed now.